packages/vue-app/template/plugins.js

```diff
@@ -0,0 +1,124 @@
+import Vue from 'vue'
+<% if (features.hookablePlugins) {
+%>import Hookable from 'hable'
+<%
+}
+
+if (features.parallelPlugins) {
+%>import { interopDefault } from './utils'
+
+<%
+  for (const pluginMode of ['all', 'server', 'client']) {
+    const modePlugins = plugins.filter(({ mode }) => mode === pluginMode)
+    if (!modePlugins.length) {
+      continue
+    }
+
+    if (pluginMode === 'all') {
+      %>const pluginImports = [
+<%
+    } else {
+      %>if (process.<%= pluginMode %>) {
+  pluginImports.concat([
+<%
+    }
+
+    modePlugins.forEach((plugin, index, array) => {
+      const sep = index === (array.length - 1) ? '' : ','
+      const space = '  '.repeat(pluginMode === 'all' ? 1 : 2)
+
+      %><%= space %>import('<%= plugin.name %>')<%= sep %> // Source: <%= relativeToBuild(plugin.src) %> (mode: '<%= plugin.mode %>')
+<%
+    })
+
+    if (pluginMode === 'all') {
+      %>]
+<%
+    } else {
+      %>  ])
+}
+<%
+    }
+  }
+} else { %>
+<%= isTest ? '/* eslint-disable camelcase */' : '' %>
+<% plugins.forEach((plugin) => { %>import <%= plugin.name %> from '<%= plugin.name %>' // Source: <%= relativeToBuild(plugin.src) %> (mode: '<%= plugin.mode %>')
+<% }) %>
+<%= isTest ? '/* eslint-enable camelcase */' : '' %>
+<% } %>
+
+export default async function executePlugins(app<%= store ? ', store' : '' %>) {
+  <% if (features.hookablePlugins) { %>
+  const hooks = new Hookable()
+  <% } %>
+
+  function inject (key, value) {
+    if (!key || value === undefined) {
+      throw new Error(`inject(key, value) has no ${key ? 'value' : 'key'} provided`)
+    }
+
+    key = `$${key}`
+    // Add into app
+    app[key] = value
+    <% if (store) { %>
+    // Add into store
+    store[key] = app[key]
+    <% } %>
+
+    if (Vue.prototype.hasOwnProperty(key)) {
```

Comment:
Could we use `in` here?
```suggestion
    if (key in Vue.prototype) {
```